include/openssl/bn.h

```diff
@@ -452,6 +452,7 @@ OPENSSL_EXPORT int GFp_BN_mod_exp_mont_consttime(BIGNUM *rr, const BIGNUM *a,
 
 /* Private functions */
 
+/* Needs to be kept in sync with `struct BIGNUM` (in `src/bn.rs`) */
```

Comment:
Your plan for creating an `Integer` that isn't constrained to be positive and that doesn't use the heap (IIUC) seems good to me. But, in the interim, an extra heap allocation where we're already doing a heap allocation doesn't matter when we know we're going to remove them all soon, so I suggest that we just KISS and jump directly to creating your `Integer` type, perhaps starting with it being a wrapper around a `BIGNUM*` and then later getting rid of its use of `BIGNUM` completely at the same time it stops using the heap.